configure.ac

```diff
@@ -1212,7 +1218,18 @@
 
 # Support for C11 atomic types
 
-OCAML_CC_SUPPORTS_ATOMIC([$cclibs])
+AS_CASE([$ocaml_cc_vendor],
+  [msvc-*],
+    [OCAML_CC_SUPPORTS_ATOMIC
+     AS_IF([! $cc_supports_atomic],
+       [OCAML_CC_SUPPORTS_ATOMIC([-std:c11])
```

Comment:
I would suggest adding `AC_MSG_NOTICE([retrying to detect atomic support with -std:c11])` (and similar for the third attempt) so that the configure log on MSVC doesn't contain three lines in a row checking for the same feature and saying no, no, yes (third time lucky?).

Reply:
Third time's the charm! I did the three checks because clang-cl doesn't need these flags, we can reasonably expect `-experimental:c11atomics` to disappear in a close future, and `-std:c11` to become unnecessary in a distant one.
I suppose I can instead alter the message in `OCAML_CC_SUPPORTS_ATOMIC` to log which flags it's trying.

Reply:
I had understood your rationale and I agree with it.
An alteration of the message displayed by `OCAML_CC_SUPPORTS_ATOMIC` would be even nicer, indeed.

Reply:
`configure` will now show for `cl`:
```
checking for cl option to support _Atomic types... unsupported
checking for cl option to support _Atomic types... unsupported
checking for cl option to support _Atomic types... -std:c11 -experimental:c11atomics
```
and will show this if the compiler supports C11 out of the box:
```
checking for cl option to support _Atomic types... none needed
```
Still, it's not exactly conveying it's retrying the check with a different set of options, but it's an improvement.

Reply:
Nice! With the small modifications in [this extra commit], I get with cl:

```
checking for cl option to support _Atomic types... failure with no option
checking for cl option to support _Atomic types... failure with -std:c11
checking for cl option to support _Atomic types... -std:c11 -experimental:c11atomics
```

and no change with clang-cl:

```
checking for clang-cl option to support _Atomic types... none needed
```

(I renamed the variable to avoid a line overflow :sweat_smile:)

[this extra commit]: https://github.com/shym/ocaml/commit/543e6decd3ca494b4aac3d6a87b11090db399259

Reply:
And it should be probably better to use a plural:
```diff
-  AC_MSG_CHECKING([for $CC option to support _Atomic types])
+  AC_MSG_CHECKING([for $CC options to support _Atomic types])
```

Reply:
Applied, many thanks!